### DISCUSSION TITLE
oidc登录https协议出现无效参数redrect_uri

### DISCUSSION THREAD

**Discussion by imhun** [hhyo/Archery#2393] [Q&A]
### 重现步骤

系统通过https访问，https通过反向代理到archery
oidc登陆时，源地址是https://xxx.com，服务端获取到的重定向地址是http://xxx.com:9123/xxx，导致出现：无效参数 redirect_uri的错误，无法正常登录

进过排查，发现是archery 内置的nginx代理没有正确配置参数导致，如下图
X-Forwarded-Host 不应该配置nginx_port，以便正确获取上游host
X-Forwarded-Proto 不应该直接取$scheme，而应该从上游代理转发的变量获取：$http_X_Forwarded_Proto
这样可以正确获取到上游的源地址https://xxx.com
<img width="636" alt="image" src="https://github.com/hhyo/Archery/assets/4082111/61b0a3ea-2f14-4f0c-865f-03ea0d082f09">
经过直接修改容器内的nginx.conf的配置，已确认问题修复

### 预期外的结果

oidc正常登录

### 版本

1.10.0

### 部署方式

Docker

**Comment by LeoQuote** [comment]
这个你可能有些误解, 请看一下这段代码 https://github.com/hhyo/Archery/blob/559791196c1c907e628a03f7c4b40ee131078b81/src/docker/startup.sh#L10-L14 

启动的时候会自动替换这个端口, 只要你不设置 nginx-port 环境变量, 就不会有这样的问题

默认的 docker-compose 是设置了这个环境变量的, 你可以把这个环境变量删除, 即可解决问题

**Comment by arthas3014** [comment]
> ### 重现步骤
> 系统通过https访问，https通过反向代理到archery oidc登陆时，源地址是[https://xxx.com，服务端获取到的重定向地址是http://xxx.com:9123/xxx，导致出现：无效参数](https://xxx.com%EF%BC%8C%E6%9C%8D%E5%8A%A1%E7%AB%AF%E8%8E%B7%E5%8F%96%E5%88%B0%E7%9A%84%E9%87%8D%E5%AE%9A%E5%90%91%E5%9C%B0%E5%9D%80%E6%98%AFhttp://xxx.com:9123/xxx%EF%BC%8C%E5%AF%BC%E8%87%B4%E5%87%BA%E7%8E%B0%EF%BC%9A%E6%97%A0%E6%95%88%E5%8F%82%E6%95%B0) redirect_uri的错误，无法正常登录
> 
> 进过排查，发现是archery 内置的nginx代理没有正确配置参数导致，如下图 X-Forwarded-Host 不应该配置nginx_port，以便正确获取上游host X-Forwarded-Proto 不应该直接取$scheme，而应该从上游代理转发的变量获取：$http_X_Forwarded_Proto 这样可以正确获取到上游的源地址https://xxx.com <img alt="image" width="636" src="https://private-user-images.githubusercontent.com/4082111/283709220-61b0a3ea-2f14-4f0c-865f-03ea0d082f09.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMyNzg1MzAsIm5iZiI6MTcxMzI3ODIzMCwicGF0aCI6Ii80MDgyMTExLzI4MzcwOTIyMC02MWIwYTNlYS0yZjE0LTRmMGMtODY1Zi0wM2VhMGQwODJmMDkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQxNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MTZUMTQzNzEwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjQyM2RkN2FmNzQ5MWQ1NjBmOWJlZWEwOTA0NWRkYTU5MTAwODY1Njk5ZjVmYjUxZDI2MzQ3NDM0MDgyZGZmMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.fqNtoMytSf-59cpXec_ascvpwKEhs-xKDtPkihy5FlY"> 经过直接修改容器内的nginx.conf的配置，已确认问题修复
> 
> ### 预期外的结果
> oidc正常登录
> 
> ### 日志文本
> _No response_
> 
> ### 版本
> 1.10.0
> 
> ### 部署方式
> Docker
> 
> ### 是否还有其他可以辅助定位问题的信息？比如数据库版本等
> _No response_

你好请问你配置了proxy_set_header   X-Forwarded-Proto $http_X_Forwarded_Proto; 就可以了是吗，可是我配置后发现还是http
